src/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -207,6 +207,11 @@ See the LICENSE file in the project root for more information.
       <IlcArg Condition="$(IlcGenerateCompleteTypeMetadata) == 'true'" Include="--completetypemetadata" />
       <IlcArg Condition="$(IlcGenerateStackTraceData) == 'true'" Include="--stacktracedata" />
       <IlcArg Condition="$(RootAllApplicationAssemblies) == 'true'" Include="--rootallapplicationassemblies" />
+      <IlcArg Include="--directpinvoke:^\*$" />
```

Comment:
Yeah, I spent half an hour trying to get that working (it was my first choice). I couldn't for my life make the items conditional on OS or TargetOS. The condition would never include anything. I tried putting it in this file, in the OS specific props file (without a condition), etc.

I see there is still something wrong for Unix anyway. Since this is an unpaid weekend project, I'm going to shelve all of this until I'm less angry at MSBuild.